assets/backport/github/github.go

```diff
@@ -0,0 +1,355 @@
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package github
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/gravitational/trace"
+
+	go_github "github.com/google/go-github/v41/github"
+	"golang.org/x/oauth2"
+)
+
+type Client struct {
+	Client *go_github.Client
+	Config
+}
+
+type Config struct {
+	Token        string
+	Organization string
+	Repository   string
+	Username     string
+}
+
+// New returns a new GitHub client.
+func New(ctx context.Context, c Config) (*Client, error) {
+	err := validateConfig(c)
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+	ts := oauth2.StaticTokenSource(
+		&oauth2.Token{AccessToken: c.Token},
+	)
+	return &Client{
+		Client: go_github.NewClient(oauth2.NewClient(ctx, ts)),
+		Config: c,
+	}, nil
+}
+
+// validateConfig validates config.
+func validateConfig(c Config) error {
+	if c.Token == "" {
+		return trace.BadParameter("missing token")
+	}
+	if c.Organization == "" {
+		return trace.BadParameter("missing organization")
+	}
+	if c.Repository == "" {
+		return trace.BadParameter("missing repository")
+	}
+	if c.Username == "" {
+		return trace.BadParameter("missing username")
+	}
+	return nil
+}
+
+// Backport backports changes from backportBranchName to a new branch based
+// off baseBranchName.
+//
+// A new branch is created with the name in the format of
+// auto-backport/[baseBranchName]/[backportBranchName], and
+// cherry-picks commits onto the new branch.
+func (c *Client) Backport(ctx context.Context, baseBranchName, backportBranchName string, commits []string) (string, error) {
+	newBranchName := fmt.Sprintf("auto-backport/%s/%s", baseBranchName, backportBranchName)
+	// Create a new branch off of the target branch.
+	err := c.createBranchFrom(ctx, baseBranchName, newBranchName)
+	if err != nil {
+		return "", trace.Wrap(err)
+	}
+	fmt.Printf("Created a new branch: %s.\n", newBranchName)
+
+	// Cherry pick commits.
+	err = c.cherryPickCommitsOnBranch(ctx, newBranchName, commits)
+	if err != nil {
+		return "", trace.Wrap(err)
+	}
+	fmt.Printf("Finished cherry-picking %v commits. \n", len(commits))
+	return newBranchName, nil
+}
+
+// CreatePullRequest creates a pull request.
+func (c *Client) CreatePullRequest(ctx context.Context, baseBranch string, headBranch string, title string, body string) error {
+	autoTitle := fmt.Sprintf("[Auto Backport] %s", title)
+	newPR := &go_github.NewPullRequest{
+		Title:               &autoTitle,
+		Head:                &headBranch,
+		Base:                &baseBranch,
+		Body:                &body,
+		MaintainerCanModify: go_github.Bool(true),
+	}
+	_, _, err := c.Client.PullRequests.Create(ctx, c.Organization, c.Repository, newPR)
+	if err != nil {
+		return err
+	}
+	return nil
+}
```

Comment:
If you change your title and body to something like `Backport #1234 to branch/v8`, GitHub will automatically create links to the original PR. For example: https://github.com/gravitational/teleport/pull/9026.